packages/vscode-extension/src/providers/PreviewCodeLensProvider.ts

```diff
@@ -13,26 +13,75 @@ export class PreviewCodeLensProvider implements CodeLensProvider {
     token: CancellationToken
   ): CodeLens[] | Thenable<CodeLens[]> {
     const text = document.getText();
+    const isStory = /.*stories.?(js|ts|jsx|tsx)$/.test(document.fileName);
 
-    // imported preview
-    if (!text.includes("react-native-ide")) {
+    if (!text.includes("react-native-ide") && !isStory) {
       return [];
     }
 
-    const regex = /\bpreview\b\s*\(/g;
-    const codeLenses = [];
+    const codeLenses: CodeLens[] = [];
 
-    for (const match of text.matchAll(regex)) {
-      const line = document.lineAt(document.positionAt(match.index).line);
-      const range = new Range(line.lineNumber, 0, line.lineNumber, line.text.length);
+    if (isStory) {
+      const componentName = this.extractComponentName(text);
+      if (!componentName) {
+        return [];
+      }
+      this.addStorybookCodeLenses(text, document, codeLenses, componentName);
+    }
+    this.addPreviewCodeLenses(text, document, codeLenses);
+    return codeLenses;
+  }
+
+  extractComponentName(text: string): string | null {
+    let componentName: string | null = null;
+    const titlePropRegex = /title:\s*(['"`])(\w+)\1/;
+    const titlePropMatch = titlePropRegex.exec(text);
+    if (titlePropMatch) {
+      componentName = titlePropMatch[2];
+    } else {
+      const componentRegex = /component:\s*(\w+)/;
+      const componentMatch = componentRegex.exec(text);
+      if (componentMatch) {
+        componentName = componentMatch[1];
+      }
+    }
+    return componentName;
+  }
+
+  addStorybookCodeLenses(
+    text: string,
+    document: TextDocument,
+    codeLenses: CodeLens[],
+    componentName: string
+  ) {
+    const storyRegex = /^(?:(?!\/\/).)*export const (\w+)(?::\s*\w+\s*)? =/gm;
```

Comment:
I'd change the first dot to space so it's more obvious we want to avoid commented out exports.